### PR TITLE
enable more clangarm64 packages

### PR DESCRIPTION
**Pull request by kmilos**

### CODE REVIEW

File: mingw-w64-adol-c/PKGBUILD [hunk 1]
```diff
@@ -7,9 +7,9 @@ pkgver=2.7.2
 pkgrel=5
 pkgdesc='Automatic Differentiation of Algorithms written in C/C++ (mingw-w64)'
 arch=('any')
-mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
 depends=("${MINGW_PACKAGE_PREFIX}-boost"
-         $([[ ${MINGW_PACKAGE_PREFIX} != *-clang-* ]] || echo "${MINGW_PACKAGE_PREFIX}-openmp"))
+         $([[ ${MINGW_PACKAGE_PREFIX} != *-clang-*86* ]] || echo "${MINGW_PACKAGE_PREFIX}-openmp"))
```
Comment (MehdiChinoune) [inline mingw-w64-adol-c/PKGBUILD:12]:
```suggestion
         $([[ ${CARCH} == "aarch64" ]] || echo "${MINGW_PACKAGE_PREFIX}-openmp"))
```

File: mingw-w64-adol-c/PKGBUILD [hunk 2]
```diff
@@ -23,25 +23,30 @@ prepare() {
...
 build() {
-  [[ -d "${srcdir}/build-${MINGW_CHOST}" ]] && rm -rf "${srcdir}/build-${MINGW_CHOST}"
-  mkdir -p "${srcdir}/build-${MINGW_CHOST}" && cd "${srcdir}/build-${MINGW_CHOST}"
+  [[ -d "${srcdir}/build-${MSYSTEM}" ]] && rm -rf "${srcdir}/build-${MSYSTEM}"
+  mkdir -p "${srcdir}/build-${MSYSTEM}" && cd "${srcdir}/build-${MSYSTEM}"
 
-  if [[ ${MINGW_PACKAGE_PREFIX} == *-clang-* ]]; then
-    LDFLAGS+=" -lomp"
+  local -a extra_config
+  if [[ ${MINGW_PACKAGE_PREFIX} == *86* ]]; then
```
Comment (MehdiChinoune) [inline mingw-w64-adol-c/PKGBUILD:30]:
```suggestion
  if [[ ${CARCH} == "aarch64" ]]; then
```

File: mingw-w64-tesseract-ocr/PKGBUILD [hunk 3]
```diff
@@ -5,17 +5,18 @@ _realname=tesseract-ocr
...
-mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
 url="https://github.com/tesseract-ocr/tesseract"
 license=("Apache License 2.0")
 makedepends=("${MINGW_PACKAGE_PREFIX}-cc" "${MINGW_PACKAGE_PREFIX}-autotools")
 depends=(${MINGW_PACKAGE_PREFIX}-asciidoc
          ${MINGW_PACKAGE_PREFIX}-cairo
          ${MINGW_PACKAGE_PREFIX}-curl
          ${MINGW_PACKAGE_PREFIX}-gcc-libs
+         $([[ ${MINGW_PACKAGE_PREFIX} != *-clang-*86* ]] || echo "${MINGW_PACKAGE_PREFIX}-openmp")
```
Comment (MehdiChinoune) [inline mingw-w64-tesseract-ocr/PKGBUILD:19]:
```suggestion
         $([[ ${CARCH} == "aarch64" ]] || echo "${MINGW_PACKAGE_PREFIX}-openmp")
```

File: mingw-w64-tesseract-ocr/PKGBUILD [hunk 4]
```diff
@@ -36,13 +37,17 @@ prepare() {
...
   cp -r "${srcdir}/tesseract-${pkgver}" "${srcdir}/build-${MSYSTEM}"
-
   cd "${srcdir}/build-${MSYSTEM}"
+
   local -a extra_config
   if check_option "debug" "y"; then
     extra_config+=( --enable-debug )
   fi
 
+  if [[ ${MINGW_PACKAGE_PREFIX} == *-clang-*86* ]]; then
```
Comment (MehdiChinoune) [inline mingw-w64-tesseract-ocr/PKGBUILD:47]:
```suggestion
  if [[ ${MINGW_PACKAGE_PREFIX} != *-clang-aarch64-* ]]; then
```

**Comment by jeremyd2019** [comment]
```
  ==> Extracting sources...
    -> Extracting tesseract-data-4.1.0.tar.gz with bsdtar
  tessdata_fast-4.1.0/configs: Can't create 'tessdata_fast-4.1.0/configs'
  tessdata_fast-4.1.0/pdf.ttf: Can't create 'tessdata_fast-4.1.0/pdf.ttf'
  bsdtar: Error exit delayed from previous errors.
```

The tar file might contain symlinks.  If so, you need to add a `noextract` array with the tar filename in it: https://github.com/msys2/MINGW-packages/blob/8b2b4cabdf6abf0dc34993669453d5d305081f4f/mingw-w64-clang/PKGBUILD#L102
and extract the tar in `prepare`: https://github.com/msys2/MINGW-packages/blob/8b2b4cabdf6abf0dc34993669453d5d305081f4f/mingw-w64-clang/PKGBUILD#L120-L123

**Comment by kmilos** [comment]
> The tar file might contain symlinks.

Right you are; should hopefully be it now.

File: mingw-w64-adol-c/PKGBUILD [hunk 5]
```diff
@@ -7,9 +7,9 @@ pkgver=2.7.2
 pkgrel=5
```
Comment (jeremyd2019) [inline mingw-w64-adol-c/PKGBUILD:7]:
```suggestion
pkgrel=6
```

File: mingw-w64-tesseract-data/PKGBUILD [hunk 6]
```diff
@@ -5,31 +5,37 @@
...
-sha256sums=('cfae2d9e15887a719c995baad70c01ad8f68c0361f5f0a4f46e4aa9ed8a47120')
+source=(${_realname}-${pkgver}.tar.gz::https://github.com/tesseract-ocr/tessdata_fast/archive/${pkgver}.tar.gz)
+sha256sums=('d0e3bb6f3b4e75748680524a1d116f2bfb145618f8ceed55b279d15098a530f9')
+noextract=(${_realname}-${pkgver}.tar.gz)
+
+prepare() {
+  cd "${srcdir}"
+
+  tar -xzf "${srcdir}/${_realname}-${pkgver}.tar.gz" || true
+)
```
Comment (jeremyd2019) [inline mingw-w64-tesseract-data/PKGBUILD:38]:
```suggestion
}
```

File: mingw-w64-adol-c/PKGBUILD (hunk 5 above)
Comment (kmilos) [reply mingw-w64-adol-c/PKGBUILD:7]: Didn't bump in on purpose, there is no real/binary change to *86 builds...

**Comment by jeremyd2019** [comment]
Successful ARM64 CI run: https://github.com/msys2-arm/MINGW-packages/runs/5890682494?check_suite_focus=true

**Comment by jeremyd2019** [comment]
Oops, #11437 introduced a conflict on tesseract-data

File: mingw-w64-tesseract-data/PKGBUILD [hunk 7]
```diff
@@ -5,35 +5,36 @@
...
+tam tat tel tgk tha tir ton tur uig ukr urd uzb uzb_cyrl vie yid yor)
 
 _langs_deva=(hin mar nep san) 
 
 _realname=tesseract-data
 pkgbase=mingw-w64-${_realname}
 pkgname=($(for l in ${_langs[@]}; do echo ${MINGW_PACKAGE_PREFIX}-${_realname}-${l}; done) )
 pkgver=4.1.0
-tagver=${pkgver}
 pkgrel=1
```
Comment (jeremyd2019) [inline mingw-w64-tesseract-data/PKGBUILD:23]: You'll need to bump pkgrel now too, as you are packaging more languages

Comment (kmilos) [reply mingw-w64-tesseract-data/PKGBUILD]: Right you are 🤦

**Pull request merged**